### PR TITLE
Add new public static method `charAt` for `Str` class

### PR DESCRIPTION
**Pull request by Awilum**
### Features
* New `Str::charAt()` method to get the character at specific index.

```php
$string = "Hellö Wörld";
echo Str::charAt($string, 0); //=> H
echo Str::charAt($string, 10); //=> d
```

## Ready?

- [x] Unit tests for fixed bug/feature
- [x] In-code documentation (wherever needed)
- [x] Tests and checks all pass

### For review team

- [ ] Add changes to release notes draft in Notion
- [x] ~Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)~

### CODE REVIEW

File: src/Toolkit/Str.php [hunk 1]
```diff
@@ -297,6 +297,18 @@ public static function camel(string $value = null): string
...
+     *
+     * @param string $string String to extract the character from
+     * @param int $index Position of the character
+     * @return string
+     */
+    public static function charAt(string $string = null, int $index): string
+    {
+        return static::substr($string, $index, 1);
+    }
+
```
Comment (lukasbestle) [inline src/Toolkit/Str.php]:
```suggestion

```

**Review by lukasbestle — APPROVED** [review]

**Comment by distantnative** [comment]
@Awilum @lukasbestle I'm wondering where we reach the point where usefulness and not blowing up the Str class more intersect. There are so many string methods one could implement, but it wouldn't really be that more useful.

Really wondering if

```php
Str::charAt($string, 4)
```

is so much more helpful than doing

```php
Str::substr($string, 4, 1)
```

**Comment by Awilum** [comment]
@distantnative 
> @Awilum @lukasbestle I'm wondering where we reach the point where usefulness and not blowing up the Str class more intersect. There are so many string methods one could implement, but it wouldn't really be that more useful.
> 
> Really wondering if
> 
> ```
> Str::charAt($string, 4)
> ```
> 
> is so much more helpful than doing
> 
> ```
> Str::substr($string, 4, 1)
> ```

So, we should not have `Str` class and use native php function `mb_substr` instead of `Str::substr` ;)

```php
mb_substr($string, 4, 1);
``` 

But as I understand `Str` class and others inside the Toolkit folder is somehow a wrapper for native PHP functions + extended functionality. Toolkit should provides rich functionality to simplify project development (aka framework/stdlib). imho.

Already builtin `Str::substr` is a simple wrapper for `mb_substr` and new `charAt` is a function for a simple job: get single char from a string.

**Comment by lukasbestle** [comment]
@Awilum We discussed your PR in our team and came to the conclusion that we won't merge it because the ratio between the usefulness of the method compared to the reduced overview of the class methods and code that needs to be maintained in the future is too low.

Despite of that, thank you very much for your contributions in the last days. They have all been of very high quality. :)

**Pull request closed without merging**